### ISSUE TITLE
Compiler 'intel@19.0.5.281' has executables that are missing or are not executable:

### ISSUE THREAD

**Issue by yellowhat** [LLNL/spack#19043]
### Steps to reproduce the issue
```console
$ git clone https://github.com/spack/spack.git
$ mkdir -p /root/.spack/linux
$ cat > /root/.spack/linux/compilers.yaml <<EOL
compilers:
- compiler:
    target: x86_64
    operating_system: centos7
    modules: []
    spec: intel@19.0.5.281
    paths:
      cc: stub
      cxx: stub
      f77: stub
      fc: stub
    flags: {}
EOL
$ source spack/share/spack/setup-env.sh
$ spack install intel@19.0.5 arch=linux-centos7-x86_64
```

### Error Message
```console
$ spack --debug --stacktrace install intel@19.0.5.281 arch=linux-centos7-x86_64 
lib/spack/spack/cmd/__init__.py:122 ==> [2020-09-29-12:19:26.563439] Imported install from built-in commands
lib/spack/spack/config.py:872 ==> [2020-09-29-12:19:26.566515] Reading config file /mnt/devtest-apps/spack/etc/spack/defaults/config.yam
l
lib/spack/spack/cmd/__init__.py:122 ==> [2020-09-29-12:19:26.591613] Imported install from built-in commands
lib/spack/spack/config.py:872 ==> [2020-09-29-12:19:26.598046] Reading config file /mnt/devtest-apps/spack/etc/spack/defaults/repos.yaml
lib/spack/spack/config.py:872 ==> [2020-09-29-12:19:26.692351] Reading config file /mnt/devtest-apps/spack/etc/spack/defaults/packages.y
aml
lib/spack/spack/config.py:872 ==> [2020-09-29-12:19:26.725284] Reading config file /root/.spack/linux/compilers.yaml
lib/spack/spack/database.py:362 ==> [2020-09-29-12:19:26.747599] DATABASE LOCK TIMEOUT: 3s
lib/spack/spack/database.py:366 ==> [2020-09-29-12:19:26.747988] PACKAGE LOCK TIMEOUT: No timeout
lib/spack/spack/installer.py:992 ==> [2020-09-29-12:19:26.766818] Initializing the build queue for intel
lib/spack/spack/package.py:1427 ==> [2020-09-29-12:19:26.795069] Warning: Missing a source id for intel@19.0.5.281
lib/spack/spack/database.py:461 ==> [2020-09-29-12:19:26.805919] Removing failure marking for intel
lib/spack/spack/installer.py:939 ==> [2020-09-29-12:19:26.812458] Acquiring a write lock on intel with timeout 1e-09
lib/spack/spack/stage.py:310 ==> [2020-09-29-12:19:26.840764] Creating stage lock spack-stage-intel-19.0.5.281-kjudjtmkccm4nfyqzcylj6a6p
dsnmuz3
lib/spack/spack/installer.py:1052 ==> [2020-09-29-12:19:26.845578] Installing intel
lib/spack/spack/installer.py:347 ==> [2020-09-29-12:19:26.845980] Searching for binary cache of intel
lib/spack/spack/config.py:872 ==> [2020-09-29-12:19:26.847594] Reading config file /mnt/devtest-apps/spack/etc/spack/defaults/mirrors.ya
ml
lib/spack/spack/binary_distribution.py:897 ==> [2020-09-29-12:19:26.850954] Finding buildcaches at https://spack-llnl-mirror.s3-us-west-
2.amazonaws.com/build_cache
lib/spack/spack/stage.py:310 ==> [2020-09-29-12:19:26.851407] Creating stage lock build_cache
lib/spack/spack/fetch_strategy.py:316 ==> [2020-09-29-12:19:26.853349] Checking existence of https://spack-llnl-mirror.s3-us-west-2.amaz
onaws.com/build_cache/linux-centos7-x86_64-intel-19.0.5.281-intel-19.0.5.281-kjudjtmkccm4nfyqzcylj6a6pdsnmuz3.spec.yaml
lib/spack/spack/util/executable.py:159 ==> [2020-09-29-12:19:26.854561] '/bin/curl' '--stderr' '-' '-s' '-f' '-r' '0-0' 'https://spack-l
lnl-mirror.s3-us-west-2.amazonaws.com/build_cache/linux-centos7-x86_64-intel-19.0.5.281-intel-19.0.5.281-kjudjtmkccm4nfyqzcylj6a6pdsnmuz
3.spec.yaml'
lib/spack/spack/stage.py:484 ==> [2020-09-29-12:19:27.670265] FailedDownloadError: Failed to fetch file from URL: https://spack-llnl-mir
ror.s3-us-west-2.amazonaws.com/build_cache/linux-centos7-x86_64-intel-19.0.5.281-intel-19.0.5.281-kjudjtmkccm4nfyqzcylj6a6pdsnmuz3.spec.
yaml
lib/spack/spack/stage.py:470 ==> [2020-09-29-12:19:27.670895] Fetching from https://spack-llnl-mirror.s3-us-west-2.amazonaws.com/build_c
ache/linux-centos7-x86_64-intel-19.0.5.281-intel-19.0.5.281-kjudjtmkccm4nfyqzcylj6a6pdsnmuz3.spec.yaml failed.
lib/spack/spack/installer.py:240 ==> [2020-09-29-12:19:27.673857] No binary for intel found: installing from source
lib/spack/spack/error.py:55 ==> [2020-09-29-12:19:27.774183] Error: CompilerAccessError: Compiler 'intel@19.0.5.281' has executables tha
t are missing or are not executable: ['stub', 'stub', 'stub', 'stub']

/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py:863, in child_process:
        860            tb_string = traceback.format_exc()
        861
        862            # build up some context from the offending package so we can
  >>    863            # show that, too.
        864            package_context = get_package_context(tb)
        865
        866            build_log = None


Traceback (most recent call last):
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 846, in child_process
    setup_package(pkg, dirty=dirty)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 721, in setup_package
    set_compiler_environment_variables(pkg, build_env)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 210, in set_compiler_environment_variables
    compiler.verify_executables()
  File "/mnt/devtest-apps/spack/lib/spack/spack/compiler.py", line 333, in verify_executables
    raise CompilerAccessError(self, missing)
CompilerAccessError: Compiler 'intel@19.0.5.281' has executables that are missing or are not executable: ['stub', 'stub', 'stub', 'stub'
]
lib/spack/spack/installer.py:1354 ==> [2020-09-29-12:19:27.774839] Flagging intel as failed: CompilerAccessError: Compiler 'intel@19.0.5
.281' has executables that are missing or are not executable: ['stub', 'stub', 'stub', 'stub']
/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py:863, in child_process:
        860            tb_string = traceback.format_exc()
        861
        862            # build up some context from the offending package so we can
  >>    863            # show that, too.
        864            package_context = get_package_context(tb)
        865
        866            build_log = None

Traceback (most recent call last):
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 846, in child_process
    setup_package(pkg, dirty=dirty)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 721, in setup_package
    set_compiler_environment_variables(pkg, build_env)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 210, in set_compiler_environment_variables
    compiler.verify_executables()
  File "/mnt/devtest-apps/spack/lib/spack/spack/compiler.py", line 333, in verify_executables
    raise CompilerAccessError(self, missing)
CompilerAccessError: Compiler 'intel@19.0.5.281' has executables that are missing or are not executable: ['stub', 'stub', 'stub', 'stub'
]

lib/spack/spack/main.py:765 ==> [2020-09-29-12:19:27.792002] ChildError: CompilerAccessError: Compiler 'intel@19.0.5.281' has executable
s that are missing or are not executable: ['stub', 'stub', 'stub', 'stub']
/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py:863, in child_process:
        860            tb_string = traceback.format_exc()
        861
        862            # build up some context from the offending package so we can
  >>    863            # show that, too.
        864            package_context = get_package_context(tb)
        865
        866            build_log = None

Traceback (most recent call last):
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 846, in child_process
    setup_package(pkg, dirty=dirty)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 721, in setup_package
    set_compiler_environment_variables(pkg, build_env)
  File "/mnt/devtest-apps/spack/lib/spack/spack/build_environment.py", line 210, in set_compiler_environment_variables
    compiler.verify_executables()
  File "/mnt/devtest-apps/spack/lib/spack/spack/compiler.py", line 333, in verify_executables
    raise CompilerAccessError(self, missing)
CompilerAccessError: Compiler 'intel@19.0.5.281' has executables that are missing or are not executable: ['stub', 'stub', 'stub', 'stub'
]
```

### Information on your system
```
$ cat /root/.spack/linux/compilers.yaml 
compilers:
- compiler:
    target: x86_64
    operating_system: centos7
    modules: []
    spec: intel@19.0.5.281
    paths:
      cc:  stub
      cxx: stub
      f77: stub
      fc:  stub
    flags: {}
```

**Comment by yellowhat** [comment]
I get the same error with tag `0.15.4` but seems to work with `v0.14.2`.
Did something change between releases?

**Comment by greschd** [comment]
I got the same error, and managed to work around it by leaving out the `stub`, i.e. changing `compilers.yaml` to 

```yaml
compilers:
- compiler:
    target: x86_64
    operating_system: centos7
    modules: []
    spec: intel@19.0.5.281
    paths:
      cc:
      cxx:
      f77:
      fc:
    flags: {}
```

This seems like it could be intentional, but in any case the Intel package documentation needs updating.

**Comment by alalazo** [comment]
Fixed in #22248
